core/src/apps/cardano/seed.py

```diff
@@ -1,11 +1,15 @@
 import storage
-import storage.cache
 from trezor import wire
 from trezor.crypto import bip32
 
 from apps.cardano import CURVE, SEED_NAMESPACE
 from apps.common import mnemonic
-from apps.common.request_passphrase import protect_by_passphrase
+from apps.common.passphrase import get as get_passphrase
+
+if False:
+    from typing import Optional
+
+_cached_root = None  # type: Optional[bytes]
```

Comment:
as I just noticed: this variable lives exactly as long as `apps.cardano.seed` is imported

that means that either (a) the unimport mechanism is working and `apps.cardano.seed` goes away after every workflow - meaning that Cardano users would need to enter passphrase before each and every operation

or (b) it survives, which would be a bug in itself -- but then doesn't respect session management and stays the same over session and passphrase changes

Reply:
It seems (a) is correct.

I like the idea you have suggested above (`storage.cache.set(key, value)`), I will implement that now.

Reply:
Check out https://github.com/trezor/trezor-firmware/pull/803/commits/d112c2d266d9d0ffbb6155bd6e7610c199be16d4